uwblueprint/jump-math
# fix: handle test session references on class archive (4)

## Notion ticket link

[Handle test sessions on class archive](https://www.notion.so/uwblueprintexecs/Handle-test-sessions-on-class-archive-29c372b5c77f4ee5868a877454621718?pvs=4)



## Implementation description
* On archive
   * Delete upcoming test sessions
   * Auto close active sessions → set end date to now
   * Return id (otherwise we need to refetch the updated class, and we don't need that info)
* Update archive modal to tell user that active sessions will end



## Steps to test
1. Create active and upcoming test sessions
<img width="1385" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/dbad13d4-886a-4229-a668-06640760bc73">

2. Archive class
<img width="630" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/135ab1be-9b01-49ac-a246-651ee826ead2">

3. Check that active test session is past and upcoming test session is deleted
<img width="1387" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/2c101359-2923-4d71-bd32-16a2712b1205">



## What should reviewers focus on?
* 


## Checklist
